package.json

```diff
@@ -36,23 +35,31 @@
     "jsdoc-fresh": "^1.0.2",
     "linkinator": "^1.4.0",
     "mocha": "^6.1.4",
-    "prettier": "^1.17.0"
+    "gts": "^0.9.0",
+    "prettier": "^1.17.0",
+    "typescript": "~3.5.3"
   },
   "scripts": {
+    "predocs": "npm run compile",
     "docs": "jsdoc -c .jsdoc.js",
-    "fix": "eslint . --fix",
-    "lint": "eslint .",
-    "test": "c8 mocha test/*.js",
+    "lint": "gts check && eslint .",
+    "clean": "gts clean",
+    "fix": "gts fix && eslint . --fix",
+    "compile": "tsc -p . && cp -r protos build/",
+    "compile-protos": "compileProtos src",
+    "prepare": "npm run compile-protos && npm run compile",
+    "pretest": "npm run compile",
+    "test": "c8 mocha build/test",
+    "system-test": "c8 mocha --timeout=15000 system-test/*.js",
+    "samples-test": "c8 mocha --timeout=800000 samples/system-test/*.js",
```

Comment:
This isn't what our samples-test command would look like.

Reply:
That was added manually, we don't generate it. We only generate these scripts:

```
  "scripts": {
    "lint": "gts check",
    "clean": "gts clean",
    "compile": "tsc -p . && cp -r protos build/",
    "compile-protos": "compileProtos src",
    "fix": "gts fix",
    "prepare": "npm run compile-protos && npm run compile",
    "test": "mocha build/test"
  },
```